### ISSUE TITLE
Soft crash when opening plugins page on >=207.7

### ISSUE THREAD

**Issue by PalmDevs** [vendetta-mod/Vendetta#192]
Vendetta - c585300  
Discord - 207.7 (207207)  
React - 18.2.0  
React Native - 0.72.3-discord-21  
Bytecode - 94

```
TypeError: Cannot read property 'on' of undefined
    at anonymous (Vendetta:1:8092)
    at commitHookEffectListMount (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:84050)
    at flushPassiveEffects (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:106387)
    at anonymous (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:102230
```

```
in ot
in Pn
in RCTView
in Unknown
in Unknown
in StaticContainer
in EnsureSingleNavigator
in SceneView
in RCTView
in Unknown
in MaybeNestedStack
in RCTView
in Unknown
in RNSScreen
in Unknown
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in InnerScreen
in AnimatedComponent(InnerScreen)
in Unknown
in Unknown
in ReanimatedScreenWrapper
in Screen
in SceneView
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in RNSScreenStack
in ScreenStack
in NativeStackViewInner
in RCTView
in Unknown
in SafeAreaProviderCompat
in NativeStackView
in PreventRemoveProvider
in NavigationContent
in Unknown
in NativeStackNavigator
in RCTView
in Unknown
in SafeAreaPadding
in SafeAreaPaddingView
in Provider
in Unknown
in Unknown
in StaticContainer
in EnsureSingleNavigator
in SceneView
in RCTView
in Unknown
in MaybeNestedStack
in RCTView
in Unknown
in RNSScreen
in Unknown
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in InnerScreen
in AnimatedComponent(InnerScreen)
in Unknown
in Unknown
in ReanimatedScreenWrapper
in Screen
in SceneView
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in RNSScreenStack
in ScreenStack
in NativeStackViewInner
in RCTView
in Unknown
in SafeAreaProviderCompat
in NativeStackView
in PreventRemoveProvider
in NavigationContent
in Unknown
in ChatPanelNativeStackNavigator
in StaticContainer
in EnsureSingleNavigator
in SceneView
in RCTView
in Unknown
in MaybeNestedStack
in RCTView
in Unknown
in RNSScreen
in Unknown
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in InnerScreen
in AnimatedComponent(InnerScreen)
in Unknown
in Unknown
in ReanimatedScreenWrapper
in Screen
in SceneView
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in RNSScreenStack
in ScreenStack
in NativeStackViewInner
in RCTView
in Unknown
in SafeAreaProviderCompat
in NativeStackView
in PreventRemoveProvider
in NavigationContent
in Unknown
in NativeStackNavigator
in Profiler
in RCTView
in Unknown
in SafeAreaPadding
in SafeAreaPaddingView
in Unknown
in RCTView
in Unknown
in AnimatedComponent(View)
in Unknown
in RCTView
in Unknown
in Wrap
in AnimatedComponent(Wrap)
in Unknown
in GestureDetector
in LaunchPadContainer
in DCDVisualEffectViewTarget
in Profiler
in StackNavigator
in PanelsViewController
in RCTView
in Unknown
in SafeAreaPadding
in SafeAreaPaddingView
in SafeAreaWrapper
in RNCSafeAreaProvider
in SafeAreaProvider
in DCDSafeArea
in DCDSafeArea
in SafeAreaProvider
in AssistantContextProvider
in KeyCommandsView
in KeyCommandsView
in ErrorBoundary
in RNGestureHandlerRootView
in GestureHandlerRootView
in GestureWrapper
in t
in WebViewContextProvider
in EnsureSingleNavigator
in BaseNavigationContainer
in ThemeProvider
in NavigationContainerInner
in AppNavigationContainer
in AppNavigationContainerOrEmpty
in RedesignIconContextProvider
in AccessibilityPreferencesContextProvider
in RootThemeContextProvider
in RootThemeContextProvider
in ReanimatedScreenProvider
in Profiler
in AppContainer
in Profiler
in profiler(AppContainer)
in Profiler
in App
in RCTView
in Unknown
in RCTView
in Unknown
in AppContainer
```

**Comment by maisymoe** [comment]
I can't seem to reproduce this at all, regardless of whether Tabs V2 is enabled or not.

**Comment by stikhead** [comment]
looks like this crash is caused by an unproxied plugin https://vendetta.nexpid.xyz/themes-plus/

**Comment by PalmDevs** [comment]
> looks like this crash is caused by an unproxied plugin https://vendetta.nexpid.xyz/themes-plus/ 

I don't use Themes+, but I'll check plugins.

**Comment by stikhead** [comment]
I checked out some plugins. After installing Cloudsync or PluginBrowser, the Plugins tab seems to work, but it stops working after you reload Discord.
The crash does not occur when below listed plugins are installed individually, but plugin tab crashes when all three of them are installed at the same time.
1. https://vd-plugins.github.io/proxy/aeongdesu.github.io/vdplugins/SlowmodeExtended/
2. https://vd-plugins.github.io/proxy/aeongdesu.github.io/vdplugins/Dashless/
3. https://vd-plugins.github.io/proxy/aeongdesu.github.io/vdplugins/FirstMessage

**Comment by PalmDevs** [comment]
@maisymoe Hey there, while you're here, can you check if clicking on a confirming action on a dialog/modal opens notification settings page after? For example, try installing a plugin.

**Comment by maisymoe** [comment]
> @maisymoe Hey there, while you're here, can you check if clicking on a confirming action on a dialog/modal opens notification settings page after? For example, try installing a plugin.

Yeah, had this pointed out to me, I think I know why.

**Comment by PalmDevs** [comment]
Update: For me it was the Experiments plugin. Not really sure why. There are also plenty of other plugins causing the exact same issue.

**Comment by speedlorenzo12** [comment]
It crashes for me even in safe mode, what should I do?

**Comment by speedlorenzo12** [comment]
It crashes for me even in safe mode, what should I do?

**Comment by PalmDevs** [comment]
It seems that Vendetta won't load any plugins (`vendetta.plugins.plugins` is an empty object) at all, if it does crash

**Comment by Red6785** [comment]
@maisymoe I'm currently getting this error too. Using CloudSync, I can export my plugins and such to a text file and upload them here. If I did this, could you check if it breaks for you as well?

**Comment by madamunkey** [comment]
I'm getting this issue using the Xposed version also, so it isn't exclusive to rootless either, if that info helps with debugging somehow

**Comment by SL4F** [comment]
> I'm getting this issue using the Xposed version also, so it isn't exclusive to rootless either, if that info helps with debugging somehow

same here, and it happened just after a fix for new Discord versions which is fun

```
TypeError: Cannot read property 'on' of undefined
    at anonymous (Vendetta:1:8092)
    at commitHookEffectListMount (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:84050)
    at flushPassiveEffects (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:106387)
    at performSyncWorkOnRoot (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:96618)
    at flushSyncCallbacks (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:31447)
    at batchedUpdatesImpl (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:118995)
    at batchedUpdates (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:16921)
    at _receiveRootNodeIDEvent (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:17404)
    at receiveTouches (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:268:18326)
    at apply (native)
    at __callFunction (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:48:4266)
    at anonymous (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:48:825)
    at __guard (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:48:2908)
    at callFunctionReturnFlushedQueue (/discord/discord_android_rn/android/app/src/main/assets/index.android.bundle:48:796)
```

```
in ot
in RCTView
in Unknown
in PostponeRender
in Unknown
in StaticContainer
in EnsureSingleNavigator
in SceneView
in RCTView
in Unknown
in RCTView
in Unknown
in RCTView
in Unknown
in CardSheet
in RCTView
in Unknown
in Unknown
in PanGestureHandler
in PanGestureHandler
in RCTView
in Unknown
in Unknown
in RCTView
in Unknown
in Card
in CardContainer
in RNSScreen
in Unknown
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in InnerScreen
in AnimatedComponent(InnerScreen)
in Unknown
in Unknown
in ReanimatedScreenWrapper
in Screen
in MaybeScreen
in RNSScreenContainer
in ScreenContainer
in MaybeScreenContainer
in RCTView
in Unknown
in Background
in CardStack
in RCTView
in Unknown
in SafeAreaProviderCompat
in RNGestureHandlerRootView
in GestureHandlerRootView
in StackView
in PreventRemoveProvider
in NavigationContent
in Unknown
in StackNavigator
in NavigationStack
in EnsureSingleNavigator
in BaseNavigationContainer
in ThemeProvider
in NavigationContainerInner
in WrappedNavigationStack
in RCTView
in Unknown
in Navigator
in Provider
in Unknown
in Unknown
in RCTView
in Unknown
in RCTView
in Unknown
in RCTView
in Unknown
in DrawerTabBar
in Profiler
in MainTabs
in Unknown
in MainRoute
in e
in e
in Unknown
in Suspender
in Suspense
in Freeze
in AppPanelFreezer
in WrappedMain
in StaticContainer
in EnsureSingleNavigator
in SceneView
in RCTView
in Unknown
in MaybeNestedStack
in RCTView
in Unknown
in RNSScreen
in Unknown
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in InnerScreen
in AnimatedComponent(InnerScreen)
in Unknown
in Unknown
in ReanimatedScreenWrapper
in Screen
in SceneView
in Suspender
in Suspense
in Freeze
in DelayedFreeze
in RNSScreenStack
in ScreenStack
in NativeStackViewInner
in RCTView
in Unknown
in SafeAreaProviderCompat
in NativeStackView
in PreventRemoveProvider
in NavigationContent
in Unknown
in NativeStackNavigator
in Profiler
in RCTView
in Unknown
in SafeAreaPadding
in SafeAreaPaddingView
in Unknown
in RCTView
in Unknown
in AnimatedComponent(View)
in Unknown
in RCTView
in Unknown
in Wrap
in AnimatedComponent(Wrap)
in Unknown
in GestureDetector
in LaunchPadContainer
in DCDVisualEffectViewTarget
in Profiler
in StackNavigator
in PanelsViewController
in RCTView
in Unknown
in SafeAreaPadding
in SafeAreaPaddingView
in SafeAreaWrapper
in RNCSafeAreaProvider
in SafeAreaProvider
in DCDSafeArea
in DCDSafeArea
in SafeAreaProvider
in AssistantContextProvider
in KeyCommandsView
in KeyCommandsView
in ErrorBoundary
in RNGestureHandlerRootView
in GestureHandlerRootView
in GestureWrapper
in t
in WebViewContextProvider
in EnsureSingleNavigator
in BaseNavigationContainer
in ThemeProvider
in NavigationContainerInner
in AppNavigationContainer
in AppNavigationContainerOrEmpty
in RedesignIconContextProvider
in AccessibilityPreferencesContextProvider
in RootThemeContextProvider
in RootThemeContextProvider
in ReanimatedScreenProvider
in Profiler
in AppContainer
in Profiler
in profiler(AppContainer)
in Profiler
in App
in RCTView
in Unknown
in RCTView
in Unknown
in AppContainer
```

**Comment by SL4F** [comment]
Vendetta - c585300
Discord - 208.0 (208200)
React - 18.2.0
React Native - 0.72.3-discord-21
Bytecode - 94

Loader - VendettaXposed
Operating System - Android 14
SDK - 34
Manufacturer - Google
Brand - google
Model - Pixel 6 Pro
Codename - raven

**Comment by Meganium97** [comment]
Can confirm this happens on stable, 207.23

**Comment by joeyoropesa-dev** [comment]
Yeah, they'ce pushed the updates from unstable channels to stable channel so now the latest stable version of Discord is not supported anymore by Vendetta. We need a fix for that first to be able to get support for higher versions.


It seems like Discord is working really hard to make modded clients impossible to keep up-to-date. They've successfuly killed Aliucord and now they are working probably on killing Vendetta too.

Sadly, but it's true. Vendetta doesn't have 20+ devs who works 24/7 to successfuly defend themself from Discord update attacks and keep their project up-to-date and functional with any update Discord pushes.

It was possible before but now with new internal plans by Discord, they will probably kill that ability for Vendetta so that if Vendetta doesn't have active 24/7 20+ devs working on this project, the project will not be able to follow updates like before.

So, what you think - should we consider making an petition for getting modded clients approved (more legal) by Discord so that Discord will not kill modded clients anymore like that with each update they made.

**Comment by Kr0n1Ku5** [comment]
Any fixes? :0

**Comment by Meganium97** [comment]
Using alpha channel. Currently fixed (?)

**Comment by Kr0n1Ku5** [comment]
@Meganium97 Back when I thought that perhaps a specific plug-in was causing the crashes, I attempted to check my plugins one by one. To see what was causing the issue.
I gave up of course after both cloud sync and platform indicators, two completely unrelated plugins lead to the same results.
So I gotta ask, are all of your plugins working properly? No compromises? You're running everything you were running before the crashes began?

**Comment by Meganium97** [comment]
Yes, I can confirm I am running basically everything that I was running pre-207.23

**Comment by PrototypeX37** [comment]
I too have the issue on 208. 

It hard crashes on booting discord. Clearing the cache works however it crashed the plugins saying cannot read property 'on' of undefined

I only ever used NoBandwidthKick

**Comment by Meganium97** [comment]
Would you mind changing your release to alpha and trying that?

**Comment by Kr0n1Ku5** [comment]
The issue has been fixed in the latest alpha.  210.5 Alpha to be more specific. There is a noticable delay in the app's startup and some of the plugin's functionality showing up properly (like a noticable increase in the time it takes to hide the gift button) but overall it seems to be completely fine and functional again.

**Comment by Meganium97** [comment]
Problem: Multiple accounts of issues still present on 210.5

Possible solution: Completely uninstall vendetta, then reinstall via vendetta manager

Discord could be hard trolling though

**Comment by Kr0n1Ku5** [comment]
@Meganium97 I had issues initially with updating a preinstalled stable build. But I just wiped user data + cache and the problem got solved.

The only issues that still persist seem to be plugin specific. I didn't mention those as I presumed, they had to be fixed by the plugin authors not the Vendetta team. 

The most apparent plugins that aren't working are things that change chat function or appearance, anything to do with hiding buttons appear buggy in appearance. Custom timestamps aren't working either. And so on.
